api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/RedHatAccountServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+    private static final String DELETING_ACCOUNT = "Deleting redhat account";
+    private static final String ADDING_ACCOUNT_SUCCESS_MESSAGE = "Account added successfully. Account id: %s";
+    private static final String REDHAT_ENABLED = "redhat.enabled";
+    private static final String SECRET_RESPONSE = "CreateSecret response: {}";
+    private static final String FORWARD_SLASH = "/";
+    private static final String SUMMARY_COUNTS_URL_PATH = "/v1/summary/counts";
+    private static final String ACCOUNT_EXISTS = "Account already exists";
+    private static final String REGION = "REGION";
+    private static final String PARAM_ACCOUNT_ID = "redhatAccountId";
+    private static final String PARAM_ACCOUNT_NAME = "redhatAccountName";
+    private static final String PARAM_TOKEN = "redhatToken";
+    private static final String DELETE_SECRET_RESPONSE = "Delete secret response: {} ";
+    private static final String ACCOUNTS_DELETED_MSG = "Redhat accounts are deleted, disabling redhat enable flag";
+    private static final String REDHAT_TOKEN_TEMPLATE = "{\"token\":\"%s\"}";
+    private static final String REDHAT_URL_TEMPLATE = "https://acs-%s.acs.rhcloud.com";
+    private static final String MISSING_MANDATORY_PARAMETER = "%s cannot be null or empty ";
+
+    @Override
+    public String serviceType() {
+        return Constants.REDHAT;
+    }
+
+    @Override
+    public AccountValidationResponse validate(CreateAccountRequest accountData) {
+        LOGGER.info(VALIDATING_REDHAT_DETAILS);
+        AccountValidationResponse accountResponse = validateRequest(accountData);
+        if (accountResponse.getValidationStatus().equalsIgnoreCase(FAILURE)) {
+            LOGGER.info(VALIDATION_FAILED_DUE_TO_MISSING_PARAMETERS);
+            return accountResponse;
+        }
+        String apiUrl = String.format(REDHAT_URL_TEMPLATE, accountData.getRedhatAccountId()) + SUMMARY_COUNTS_URL_PATH;
+        String redhatToken = accountData.getRedhatToken();
+
+        HttpClient httpClient = HttpClients.createDefault();
+        HttpGet request = new HttpGet(apiUrl);
+
+        // Set the access token in the Authorization header
+        request.setHeader(HttpHeaders.AUTHORIZATION, BEARER_PREFIX + redhatToken);
+
+        try {
+            // Make the API request
+            HttpResponse response = httpClient.execute(request);
+            // Get the HTTP status code
+            int statusCode = response.getStatusLine().getStatusCode();
+
+            // Create an AccountValidationResponse based on the status code
+            if (statusCode == 200) {
+                accountResponse.setValidationStatus(SUCCESS);
+                accountResponse.setMessage(VALIDATION_SUCCESSFUL);
+                LOGGER.info(VALIDATION_SUCCESSFUL);
+            } else {
+                accountResponse.setValidationStatus(FAILURE);
+                accountResponse.setMessage(TOKEN_VALIDATION_FAILED);
```

Comment:
how do you know if validation failed because of bad token? what if API is down?
it would be better to check a specific token failure statusCode like 401, 403, etc which relate to a token issue.
all other codes should be handled gracefully, like account created but offline?

Reply:
Im also returning status code with the response
![image](https://github.com/PaladinCloud/CE/assets/114460784/eaeea671-8fd7-4db5-aa47-698dc2993fd6)
if host is down or wrong account id, it will throw UnknownHostException and we are catching it. As we are only taking accountId as input for the URL creation, so we can say that it is due to accountId